builtin/providers/test/resource_with_custom_diff.go

```diff
@@ -0,0 +1,84 @@
+package test
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func testResourceCustomDiff() *schema.Resource {
+	return &schema.Resource{
+		Create:        testResourceCustomDiffCreate,
+		Read:          testResourceCustomDiffRead,
+		CustomizeDiff: testResourceCustomDiffCustomizeDiff,
```

Comment:
Minor cosmetic nit: visually most of the existing `Resource` attributes are short, so they line up nicely when formatted in a struct literal like this. Perhaps we could preserve that here by just calling this `Diff`. I think that's consistent with the other usage here, since _all_ of these operations are in some sense "customizing" the default behavior that `helper/schema` naturally does.